jest-circus test utilities

aaronabramov opened the pull request:
utility function that runs a test using `jest-circus` and logs every event that goes through the event handler. (see snapshot test)

This is useful to test the definition and execution flow and make sure future work won't break the flow. (e.g. `beforeEach` suddenly isn't invoked before test)

aaronabramov commented on packages/jest-util/package.json:
```diff
@@ -13,8 +13,7 @@
     "jest-message-util": "^22.4.0",
-    "mkdirp": "^0.5.1",
-    "source-map": "^0.6.0"
+    "mkdirp": "^0.5.1"
```
not sure why this and `yarn.lock` got messed up. am i using an old yarn version on something? or is master currently out of sync?

SimenB replied:
this is weird - source-map was just added in #5594. Maybe you need a rebase?

running `yarn` on master for me does not change the lockfile either

cpojer commented:
Nice!

cpojer commented on packages/jest-circus/src/__mocks__/test_utils.js:
```diff
@@ -0,0 +1,76 @@
+const CIRCUS_RUN_PATH = path.resolve(__dirname, '../../build/run');
+const CIRCUS_STATE_PATH = path.resolve(__dirname, '../../build/state');
+const TEST_EVENT_HANDLER_PATH = path.resolve(__dirname, './test_event_handler');
+const BABEL_REGISTER_PATH = path.resolve(
```
can you use `require.resolve` or `require.main.resolve` here?

aaronabramov commented:
replaced `path.resolve` with` require.resolve` and fixed the lock file. seems like a had an ancient version of yarn that updated all packages an `yarn add`

aaronabramov commented:
i'm going to skip this test suite on windows, because it seems like it has a problem with resolving some paths. This test utils should not be platform specific anyway

<img width="671" alt="screen shot 2018-02-26 at 6 37 37 pm" src="https://user-images.githubusercontent.com/940133/36707788-c27aa954-1b24-11e8-9383-24271161f805.png">

aaronabramov commented:
it might also be just a cache that i can't reset :/
<img width="495" alt="screen shot 2018-02-26 at 6 42 52 pm" src="https://user-images.githubusercontent.com/940133/36707846-ea6788d8-1b24-11e8-8510-1386dbcc24e1.png">

thymikee commented on packages/jest-circus/src/__tests__/__snapshots__/hooks.test.js.snap:
```diff
@@ -0,0 +1,64 @@
+add_hook: beforeEach
+add_test: one
+add_test: two
+start_describe_definition: 2nd level describe
```
This test would be even better if it had the depth represented with indents. Would you mind adding that?
